src/overlays/actors/ovl_En_Horse_Link_Child/z_en_horse_link_child.c

```diff
@@ -27,47 +31,558 @@ const ActorInit En_Horse_Link_Child_InitVars = {
+    Player* player = PLAYER;
+    f32 dist;
+    s32 newAnimationIdx;
+
+    this->timer++;
+    if (this->timer > 300) {
+        this->unk_1E8 = true;
+    }
+
+    if ((this->animationIdx == 4) || (this->animationIdx == 3) || (this->animationIdx == 2)) {
+        if (!this->unk_1E8) {
+            func_8006DD9C(&this->actor, &player->actor.posRot.pos, 300);
+        } else {
+            func_8006DD9C(&this->actor, &this->actor.initPosRot.pos, 300);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A695A4.s")
+    if (SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+        if (!this->unk_1E8) {
+            dist = func_8002DB8C(&this->actor, &PLAYER->actor);
+        } else {
+            dist = Math3D_Vec3f_DistXYZ(&this->actor.posRot.pos, &this->actor.initPosRot.pos);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/EnHorseLinkChild_Init.s")
+        if (!this->unk_1E8) {
+            if (dist >= 300.0f) {
+                newAnimationIdx = 4;
+                this->actor.speedXZ = 6.0f;
+            } else if (dist >= 150.0f) {
+                newAnimationIdx = 3;
+                this->actor.speedXZ = 4.0f;
+            } else {
+                newAnimationIdx = 2;
+                this->actor.speedXZ = 2.0f;
+                this->unk_1F0 = 0;
+            }
+        } else {
+            if (dist >= 300.0f) {
+                newAnimationIdx = 4;
+                this->actor.speedXZ = 6.0f;
+            } else if (dist >= 150.0f) {
+                newAnimationIdx = 3;
+                this->actor.speedXZ = 4.0f;
+            } else if (dist >= 70.0f) {
+                newAnimationIdx = 2;
+                this->actor.speedXZ = 2.0f;
+                this->unk_1F0 = 0;
+            } else {
+                func_80A6A4DC(this);
+                return;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/EnHorseLinkChild_Destroy.s")
+        if (this->animationIdx != newAnimationIdx) {
+            this->animationIdx = newAnimationIdx;
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A695A4(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, -5.0f);
+        } else {
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A695A4(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, 0.0f);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Link_Child/func_80A6988C.s")
+static EnHorseLinkChildActionFunc D_80A6AF7C[] = { func_80A698F4, func_80A69C18, func_80A699FC,
+                                                   func_80A6A068, func_80A6A7D0, func_80A6A5A4 };
```

Comment:
you can put a comma on the last entry to stop the indentation. and you can call it `sActionFuncs` if you want